File: find-the-town-judge/find-the-town-judge.go
// https://leetcode.com/problems/find-the-town-judge/
// 96 ms 7.3 MB
package main

func findJudge(n int, trust [][]int) int {
	counter := make([]int, n)
	for i := range trust {
		counter[trust[i][1]-1]++
		counter[trust[i][0]-1]--
	}
	for i := range counter {
		if counter[i] == n-1 {
			return i + 1
		}
	}
	return -1
}

File: find-the-town-judge/find-the-town-judge.c
// https://leetcode.com/problems/find-the-town-judge/
// 178 ms 16.3 MB
#include <stdio.h>
#include <stdlib.h>

int findJudge(int n, int** trust, int trustSize, int* trustColSize) {
    int i;
    int* counter = calloc(n, sizeof(int));
    for (i = 0; i < trustSize; i++) {
        counter[trust[i][1] - 1]++;
        counter[trust[i][0] - 1]--;
    }
    for (i = 0; i < n; i++) {
        if (counter[i] == n - 1) {
            // free(counter);
            return i + 1;
        }
    }
    // free(counter);
    return -1;
}